sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.cryptography;
+
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.BaseBlobClientBuilder;
+import com.azure.storage.blob.BlobURLParts;
+import com.azure.storage.blob.BlockBlobAsyncClient;
+import com.azure.storage.blob.BlockBlobClient;
+import com.azure.storage.blob.URLParser;
+import com.azure.storage.blob.implementation.AzureBlobStorageBuilder;
+import com.azure.storage.blob.implementation.AzureBlobStorageImpl;
+import com.azure.storage.blob.models.CustomerProvidedKey;
+import com.azure.storage.common.credentials.SASTokenCredential;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class provides a fluent builder API to help aid the configuration and instantiation Storage Blob clients.
+ *
+ * <p>
+ * The following information must be provided on this builder:
+ *
+ * <ul>
+ * <li>the endpoint through {@code .endpoint()}, including the container name and blob name, in the format of
```

Comment:
>including the container name and blob name [](start = 49, length = 42)

"if the container and blob name are not specified, they must be specified through {@code .blobName} and {@code .containerName}."